### PR TITLE
Add weight of labels for NER task

### PR DESCRIPTION
**Pull request by tiena2cva**
When I worked with some NER-like task, I want to use weighted labels but the origin version of simpletransformers is not support that feature.
- Add weight parameter when create NER model
- Calculate loss with weight

### CODE REVIEW

File: simpletransformers/ner/ner_model.py [hunk 1]
```diff
@@ -704,10 +718,16 @@ def train(
                         outputs = model(**inputs)
                         # model outputs are always tuple in pytorch-transformers (see doc)
                         loss = outputs[0]
+                        if self.loss_fct:
```
Comment (Zhylkaaa) [inline simpletransformers/ner/ner_model.py:721]: maybe consider filtering some of this label, because not all tokens are taken into account during loss calculation (I mean padding etc)
Maybe something like original implementation?
```active_loss = attention_mask.view(-1) == 1
                active_logits = logits.view(-1, self.num_labels)
                active_labels = torch.where(
                    active_loss, labels.view(-1), torch.tensor(loss_fct.ignore_index).type_as(labels)
                )
                loss = loss_fct(active_logits, active_labels)```
https://github.com/huggingface/transformers/blob/master/src/transformers/models/roberta/modeling_roberta.py#L1402

File: simpletransformers/ner/ner_model.py [hunk 2]
```diff
@@ -704,10 +718,16 @@ def train(
...
+                            logits = outputs[1]
+                            loss = self.loss_fct(logits.view(-1, self.num_labels), inputs['labels'].view(-1))
```
Comment (tiena2cva) [inline simpletransformers/ner/ner_model.py:723]:
```suggestion
                        if self.loss_fct:
                            logits = outputs[1]
                            labels = inputs['labels']
                            if inputs.get('attention_mask') is not None:
                                active_loss = attention_mask.view(-1) == 1
                                active_logits = logits.view(-1, self.num_labels)
                                active_labels = torch.where(
                                    active_loss, labels.view(-1), torch.tensor(loss_fct.ignore_index).type_as(labels)
                                )
                                loss = loss_fct(active_logits, active_labels)
                            else:
                                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
```

@Zhylkaaa I understand and I think you are right. Look at the code in the suggestion, are you ok with that?

But I don't think this pull request will be merge because it was opened a long time ago and noone reviews or comment.

File: simpletransformers/ner/ner_model.py (hunk 1 above)
Comment (tiena2cva) [reply simpletransformers/ner/ner_model.py:721]: answer in [review](#pullrequestreview-702707312)

File: simpletransformers/ner/ner_model.py (hunk 2 above)
Comment (Zhylkaaa) [reply simpletransformers/ner/ner_model.py:723]: yes, I think this will work. Also I hope that this changes will be merged :) 
There are still some commits to master so I guess library is alive 🤞

Comment (ThilinaRajapakse) [reply simpletransformers/ner/ner_model.py]: Sorry this took so long! I'll merge this shortly.

**Review by ThilinaRajapakse — APPROVED** [review]

**Comment by ThilinaRajapakse** [comment]
@allcontributors add @tiena2cva for code

**Comment by Zhylkaaa** [comment]
I have added #1191 fixing minor bugs and refactoring code
